backend/src/services/background/background.service.ts

```diff
@@ -0,0 +1,38 @@
+import { instance as InstanceRep } from '~/data/repositories/repositories';
+import { ec2 as EC2Serv, instance as instanceServ } from '~/services/services';
+import { ENV } from '~/common/enums/enums';
+
+type Constructor = {
+  instanceService: typeof instanceServ;
+  instanceRepository: typeof InstanceRep;
+  ec2Service: typeof EC2Serv;
+};
+
+class Background {
+  #instanceService: typeof instanceServ;
+  #instanceRepository: typeof InstanceRep;
+  #ec2Service: typeof EC2Serv;
+
+  constructor({
+    instanceService,
+    instanceRepository,
+    ec2Service,
+  }: Constructor) {
+    this.#instanceService = instanceService;
+    this.#instanceRepository = instanceRepository;
+    this.#ec2Service = ec2Service;
+  }
+
+  public async backgroundWork(): Promise<void> {
+    const { FLAGS } = ENV;
+    FLAGS.INSTANCE_AUTO_DELETING
+      ? setInterval(async () => {
+          const oldInstances = await this.#instanceRepository.getOldInstances();
+
+          oldInstances.forEach((instance) => instanceServ.delete(instance.id));
+        }, 60 * 60 * 1000)
+      : null;
+  }
+}
```

Comment:
Having all background jobs in one method is a really bad idea. What would you do if you had 10 different jobs? I like an approach when one file = one background job. So `class Background` => `class DeleteOldInstancesaBckgroundJob`